core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/DiskUpdateController.java

```diff
@@ -0,0 +1,49 @@
+package com.sequenceiq.cloudbreak.controller.v4;
+
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.validation.constraints.NotEmpty;
+
+import org.springframework.stereotype.Controller;
+
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.DiskUpdateEndpoint;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.request.DiskUpdateRequest;
+import com.sequenceiq.cloudbreak.api.endpoint.v4.stacks.request.SdxDiskUpdateRequest;
+import com.sequenceiq.cloudbreak.cloud.model.Volume;
+import com.sequenceiq.cloudbreak.service.datalake.DiskUpdateService;
+import com.sequenceiq.flow.api.model.FlowIdentifier;
+
+@Controller
+public class DiskUpdateController implements DiskUpdateEndpoint {
+
+    @Inject
+    private DiskUpdateService diskUpdateService;
+
+    @Override
+    public Boolean isDiskTypeChangeSupported(@NotEmpty String platform) {
+        return diskUpdateService.isDiskTypeChangeSupported(platform);
+    }
+
+    @Override
+    public void updateDiskTypeAndSize(SdxDiskUpdateRequest sdxDiskUpdateRequest) throws Exception {
+        DiskUpdateRequest diskUpdateRequest = sdxDiskUpdateRequest.getDiskUpdateRequest();
+        List<Volume> volumesToUpdate = sdxDiskUpdateRequest.getVolumesToUpdate();
+        diskUpdateService.updateDiskTypeAndSize(diskUpdateRequest, volumesToUpdate, sdxDiskUpdateRequest.getStackId());
+    }
+
+    @Override
+    public void stopCMServices(long stackId) throws Exception {
```

Comment:
I cannot tell if you intend to use this for datahub. if yes and this is not an internal api, then please use resource crn or name instead of id. if this is internal api then please corresponding parameters to make it possible to call it internally. this applies for all of the methods in the controller